src/compiler/checker.ts

```diff
@@ -26429,7 +26448,10 @@ namespace ts {
             // accessor, or optional method.
             const assignmentKind = getAssignmentTargetKind(node);
             if (assignmentKind === AssignmentKind.Definite ||
-                prop && !(prop.flags & (SymbolFlags.Variable | SymbolFlags.Property | SymbolFlags.Accessor)) && !(prop.flags & SymbolFlags.Method && propType.flags & TypeFlags.Union)) {

```

Comment:
Only change here is to exempt `isDuplicatedCommonJSExport` so that it falls through to the autoType case below